x/supply/coinGeckoTypes/coins.go

```diff
@@ -0,0 +1,41 @@
+package coingecko
+
+// Coin this represent coingecko api's coins/list attributes
+type Coin struct {
+	ID     string `json:"id"`
+	Symbol string `json:"symbol"`
+	Name   string `json:"name"`
+}
+
+// NewCoin return a new coingecko coin instance
+func NewCoin(id string, symbol string, name string) Coin {
+	return Coin{
+		ID:     id,
+		Symbol: symbol,
+		Name:   name,
+	}
+}
+
+// Coins is an array of Coin
+type Coins []Coin
+
+//____________________________________________________
+
+// Market represent some of the attributes in coingecko api's coins/market
+type Market struct {
```

Comment:
I would probably rename this to be `MarketTicker` or something else more specific. Right now it doesn't allow to understand exactly what it represents